R/chop.R

```diff
@@ -37,9 +37,9 @@
 #'   will be dropped from the output. If you want to preserve all rows,
 #'   use `keep_empty = TRUE` to replace size-0 elements with a single row
 #'   of missing values.
-#' @param ptype Optionally, supply a data frame prototype for the output `cols`,
-#'   overriding the default that will be guessed from the combination of
-#'   individual values.
+#' @param ptype Optionally, a named list of column name-prototype pairs to
+#'   coerce `cols` to, overriding the default that will be guessed from
+#'   combining the individual values.
```

Comment:
One subtle change is that `ptype` no longer has to be a data frame. It is now just a named list where the names can match any names specified by `cols`. This matches `pivot_longer()`.

Previously we required a data frame here, but we had some strange behavior where we would cast the unchopped data frame to `ptype`, which could add columns that only ever existed in the ptype. That seemed very strange, and this feels like much better behavior for this argument.